src/test/java/CalendarServletTest.java

```diff
@@ -0,0 +1,140 @@
+      ImmutableList.of(
+          new Event().setSummary(EVENT_SUMMARY_ONE), new Event().setSummary(EVENT_SUMMARY_TWO));
+  private static final List<Event> EVENT_UNDEFINED = ImmutableList.of(new Event());
+  private static final List<Event> EVENT_ALL =
+      ImmutableList.of(
+          new Event(),
+          new Event().setSummary(EVENT_SUMMARY_ONE),
+          new Event().setSummary(EVENT_SUMMARY_TWO));
+
+  @Before
+  public void setUp() throws IOException, GeneralSecurityException {
+    authenticationVerifier = Mockito.mock(AuthenticationVerifier.class);
+    calendarClientFactory = Mockito.mock(CalendarClientFactory.class);
+    calendarClient = Mockito.mock(CalendarClient.class);
+    servlet = new CalendarServlet(authenticationVerifier, calendarClientFactory);
+
+    Mockito.when(calendarClientFactory.getCalendarClient(Mockito.any())).thenReturn(calendarClient);
+    // Authentication will always pass
+    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
+        .thenReturn(AUTHENTICATION_VERIFIED);
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    Mockito.when(request.getCookies()).thenReturn(validCookies);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @Test
+  public void noCalendarEvent() throws IOException, ServletException {
+    // Test case where there are no events in the user's calendar
+    Mockito.when(calendarClient.getCalendarEvents()).thenReturn(NO_EVENT);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+  }
+
+  @Test
+  public void firstTwoEvents() throws IOException, ServletException {
+    // Test case where there are two events with defined summaries
+    Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_ONE_TWO);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EVENT_ONE_TWO_JSON));
+  }
+
+  @Test
+  public void undefinedEvent() throws IOException, ServletException {
+    // Test case where there is an event with no summary
+    Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_UNDEFINED);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EVENT_UNDEFINED_JSON));
+  }
+
+  @Test
+  public void allEvent() throws IOException, ServletException {
+    // Test case where there are two defined and an undefined event
+    Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_ALL);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EVENT_ALL_JSON));
+  }
```

Comment:
nit: It might help to decode the JSON: https://github.com/googleinterns/step7-2020/pull/40#discussion_r453738090

Reply:
Agreed, I changed the tests to decode instead